### PR TITLE
Adapt Application Label API to use Formations API

### PR DESCRIPTION
**Pull request by ivantenevvasilev**
**Description**

Changes proposed in this pull request:
- adapt `registerApplication` to use Formations API for scenario label
- adapt `setApplicationLabel` to use Formations API for scenario label
- adapt `deleteApplicationLabel` to use Formations API for scenario label
- adapt old unit tests
- introduce new unit tests to cover new cases

**Related issue(s)**
- ...

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] `chart/compass/values.yaml` is updated
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Review by dzahariev — DISMISSED** [review]

**Review by StanislavStefanov — DISMISSED** [review]

File: components/director/internal/domain/application/service.go [hunk 1]
```diff
@@ -705,7 +720,15 @@ func (s *service) DeleteLabel(ctx context.Context, applicationID string, key str
 		return errors.Wrap(err, "while checking Application existence")
 	}
 	if !appExists {
-		return fmt.Errorf("application with ID %s doesn't exist", applicationID)
+		return errors.Errorf("application with ID %s doesn't exist", applicationID)
+	}
+
+	if key == model.ScenariosKey {
+		scenarios, err := label.ValueToStringsSlice(labelValue)
```
[the remark this answers is not in the document]
Comment (ivantenevvasilev) [reply components/director/internal/domain/application/service.go:727]: nice catch

**Review by dzahariev — APPROVED** [review]

File: components/director/internal/domain/application/service_test.go [hunk 2]
```diff
@@ -5006,10 +4987,11 @@ func TestService_DeleteLabel(t *testing.T) {
 		t.Run(testCase.Name, func(t *testing.T) {
 			repo := testCase.RepositoryFn()
 			labelRepo := testCase.LabelRepositoryFn()
-			svc := application.NewService(nil, nil, repo, nil, nil, labelRepo, nil, nil, nil, nil, nil)
+			formationSvc := testCase.FormationServiceFn()
+			svc := application.NewService(nil, nil, repo, nil, nil, labelRepo, nil, nil, nil, nil, nil, formationSvc)
 
 			// WHEN
-			err := svc.DeleteLabel(ctx, testCase.InputApplicationID, testCase.InputKey)
+			err := svc.DeleteLabel(ctx, testCase.InputApplicationID, testCase.InputKey) // TODO
```
Comment (StanislavStefanov) [inline components/director/internal/domain/application/service_test.go:4994]: leftover

File: components/director/internal/domain/application/service_test.go [hunk 3]
```diff
@@ -4992,12 +4920,65 @@ func TestService_DeleteLabel(t *testing.T) {
...
 				return repo
 			},
+			LabelRepositoryFn:  UnusedLabelRepository,
+			FormationServiceFn: UnusedFormationService,
+			InputApplicationID: applicationID,
+			InputKey:           labelKey,
+			ExpectedErrMessage: testErr.Error(),
+		},
+		{
+			Name: "Returns error when application succeeds but application does not exist",
```
Comment (StanislavStefanov) [inline components/director/internal/domain/application/service_test.go:4930]: "Returns error when application does not exist" ?

**Review by StanislavStefanov — CHANGES_REQUESTED** [review]

**Review by StanislavStefanov — APPROVED** [review]

**Review by StanislavStefanov — APPROVED** [review]

**Review by dzahariev — APPROVED** [review]

**Review by dzahariev — APPROVED** [review]

**Pull request merged**